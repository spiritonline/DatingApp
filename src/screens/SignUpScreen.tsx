import { useState } from 'react';
import { useColorScheme, View, Alert, ScrollView, KeyboardAvoidingView, Platform } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { createUserWithEmailAndPassword } from 'firebase/auth';
import { auth } from '../services/firebase';
import { AuthNavigationProp } from '../navigation/types';
import styled from 'styled-components/native';
import { SafeAreaView } from 'react-native-safe-area-context';
import AuthHeader from '../components/auth-components/AuthHeader';
import { validateEmail, validatePassword } from '../utils/validation';

export default function SignUpScreen() {
  const colorScheme = useColorScheme();
  const navigation = useNavigation<AuthNavigationProp>();
  const isDark = colorScheme === 'dark';

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [hasAcceptedTerms, setHasAcceptedTerms] = useState(false);
  const [hasAcceptedLocationChecks, setHasAcceptedLocationChecks] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [errors, setErrors] = useState({
    email: '',
    password: '',
    confirmPassword: '',
    terms: '',
  });

  const validateForm = () => {
    let isValid = true;
    const newErrors = {
      email: '',
      password: '',
      confirmPassword: '',
      terms: '',
    };

    // Email validation with comprehensive checks
    if (!email) {
      newErrors.email = 'Email is required';
      isValid = false;
    } else if (!validateEmail(email)) {
      newErrors.email = 'Please enter a valid email address';
      isValid = false;
    }

    // Password validation with strength checking
    if (!password) {
      newErrors.password = 'Password is required';
      isValid = false;
    } else {
      const passwordValidation = validatePassword(password);
      if (!passwordValidation.valid) {
        newErrors.password = passwordValidation.suggestions[0] || 'Password is invalid';
        isValid = false;
      } else if (passwordValidation.strength === 'weak') {
        // Show suggestions for weak passwords
        newErrors.password = 'Weak password: ' + passwordValidation.suggestions.join(', ');
        isValid = false;
      }
    }

    // Confirm password validation
    if (password !== confirmPassword) {
      newErrors.confirmPassword = 'Passwords do not match';
      isValid = false;
    }

    // Terms validation
    if (!hasAcceptedTerms || !hasAcceptedLocationChecks) {
      newErrors.terms = 'You must accept the terms and location checks';
      isValid = false;
    }

    setErrors(newErrors);
    return isValid;
  };

  const handleSignUp = async () => {
    if (!validateForm()) return;

    setIsLoading(true);
    try {
      await createUserWithEmailAndPassword(auth, email, password);
      navigation.navigate('MainFeed');
    } catch (error: any) {
      console.error('Error signing up:', error);
      Alert.alert(
        'Sign Up Failed',
        error.message || 'Failed to create account. Please try again.'
      );
    } finally {
      setIsLoading(false);
    }
  };

  const handleGoogleSignUp = () => {
    // Implementation would go here in a real app
    Alert.alert('Google Sign Up', 'Google Sign Up would be implemented here');
  };

  const handleAppleSignUp = () => {
    // Implementation would go here in a real app
    Alert.alert('Apple Sign Up', 'Apple Sign Up would be implemented here');
  };

  return (
    <Container isDark={isDark}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={{ flex: 1, width: '100%' }}
      >
        <ScrollView 
          contentContainerStyle={{ flexGrow: 1 }}
          keyboardShouldPersistTaps="handled"
        >
          <AuthHeader title="Create Account" onBack={() => navigation.goBack()} />
          <FormContainer>
            <InputContainer>
              <InputLabel isDark={isDark}>Email</InputLabel>
              <Input
                isDark={isDark}
                value={email}
                onChangeText={setEmail}
                placeholder="Enter your email"
                placeholderTextColor="#999"
                keyboardType="email-address"
                autoCapitalize="none"
                autoComplete="email"
                testID="email-input"
                accessibilityLabel="Email input"
                accessibilityHint="Enter your email address"
              />
              {errors.email ? <ErrorText>{errors.email}</ErrorText> : null}
            </InputContainer>

            <InputContainer>
              <InputLabel isDark={isDark}>Password</InputLabel>
              <Input
                isDark={isDark}
                value={password}
                onChangeText={setPassword}
                placeholder="Create a password"
                placeholderTextColor="#999"
                secureTextEntry
                testID="password-input"
                accessibilityLabel="Password input"
                accessibilityHint="Create a password with at least 6 characters"
              />
              {errors.password ? <ErrorText>{errors.password}</ErrorText> : null}
            </InputContainer>

            <InputContainer>
              <InputLabel isDark={isDark}>Confirm Password</InputLabel>
              <Input
                isDark={isDark}
                value={confirmPassword}
                onChangeText={setConfirmPassword}
                placeholder="Confirm your password"
                placeholderTextColor="#999"
                secureTextEntry
                testID="confirm-password-input"
                accessibilityLabel="Confirm password input"
                accessibilityHint="Confirm your password"
              />
              {errors.confirmPassword ? <ErrorText>{errors.confirmPassword}</ErrorText> : null}
            </InputContainer>

            <CheckboxContainer>
              <Checkbox
                accessibilityRole="checkbox"
                accessibilityState={{ checked: hasAcceptedTerms }}
                onPress={() => setHasAcceptedTerms(!hasAcceptedTerms)}
                testID="terms-checkbox"
              >
                <CheckboxBox isChecked={hasAcceptedTerms} isDark={isDark}>
                  {hasAcceptedTerms && <CheckboxCheck>✓</CheckboxCheck>}
                </CheckboxBox>
                <CheckboxLabel isDark={isDark}>
                  I accept the Terms of Service and Privacy Policy
                </CheckboxLabel>
              </Checkbox>
            </CheckboxContainer>

            <CheckboxContainer>
              <Checkbox
                accessibilityRole="checkbox"
                accessibilityState={{ checked: hasAcceptedLocationChecks }}
                onPress={() => setHasAcceptedLocationChecks(!hasAcceptedLocationChecks)}
                testID="location-checkbox"
              >
                <CheckboxBox isChecked={hasAcceptedLocationChecks} isDark={isDark}>
                  {hasAcceptedLocationChecks && <CheckboxCheck>✓</CheckboxCheck>}
                </CheckboxBox>
                <CheckboxLabel isDark={isDark}>
                  I consent to 3-day location checks
                </CheckboxLabel>
              </Checkbox>
            </CheckboxContainer>

            {errors.terms ? <ErrorText>{errors.terms}</ErrorText> : null}

            <SignUpButton 
              onPress={handleSignUp} 
              disabled={isLoading}
              testID="signup-button"
              accessibilityRole="button"
              accessibilityLabel="Sign Up Button"
              accessibilityHint="Create a new account with the provided information"
            >
              <ButtonText>{isLoading ? 'Creating Account...' : 'Sign Up'}</ButtonText>
            </SignUpButton>

            <Divider isDark={isDark}>
              <DividerLine isDark={isDark} />
              <DividerText isDark={isDark}>or sign up with</DividerText>
              <DividerLine isDark={isDark} />
            </Divider>

            <SocialButtonsContainer>
              <SocialButton 
                onPress={handleGoogleSignUp}
                testID="google-signup-button"
                accessibilityRole="button"
                accessibilityLabel="Sign up with Google"
              >
                <SocialButtonText isDark={isDark}>Google</SocialButtonText>
              </SocialButton>

              <SocialButton 
                onPress={handleAppleSignUp}
                testID="apple-signup-button"
                accessibilityRole="button"
                accessibilityLabel="Sign up with Apple"
              >
                <SocialButtonText isDark={isDark}>Apple</SocialButtonText>
              </SocialButton>
            </SocialButtonsContainer>

            <BottomText>
              Already have an account?{' '}
              <LinkText 
                onPress={() => navigation.navigate('LogIn')}
                accessibilityRole="link"
                accessibilityLabel="Log In"
                accessibilityHint="Navigate to log in screen"
              >
                Log In
              </LinkText>
            </BottomText>
          </FormContainer>
        </ScrollView>
      </KeyboardAvoidingView>
    </Container>
  );
}

import { ThemeProps, CheckboxProps } from '../utils/styled-components';

const Container = styled(SafeAreaView)<ThemeProps>`
  flex: 1;
  background-color: ${(props: ThemeProps) => (props.isDark ? '#121212' : '#ffffff')};
`;

const FormContainer = styled.View`
  flex: 1;
  padding: 20px;
  align-items: center;
`;

const InputContainer = styled.View`
  width: 100%;
  margin-bottom: 16px;
`;

const InputLabel = styled.Text<ThemeProps>`
  font-size: 14px;
  margin-bottom: 8px;
  color: ${(props: ThemeProps) => (props.isDark ? '#e0e0e0' : '#333')};
`;

const Input = styled.TextInput<ThemeProps>`
  width: 100%;
  padding: 16px;
  border-radius: 8px;
  background-color: ${(props: ThemeProps) => (props.isDark ? '#2c2c2c' : '#f5f5f5')};
  color: ${(props: ThemeProps) => (props.isDark ? '#ffffff' : '#000000')};
  border: 1px solid ${(props: ThemeProps) => (props.isDark ? '#444' : '#ddd')};
`;

const ErrorText = styled.Text`
  color: #ff3b30;
  font-size: 12px;
  margin-top: 4px;
`;

const CheckboxContainer = styled.View`
  width: 100%;
  margin-bottom: 12px;
`;

const Checkbox = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
`;

const CheckboxBox = styled.View<CheckboxProps>`
  width: 20px;
  height: 20px;
  border-radius: 4px;
  border: 1px solid ${(props: CheckboxProps) => (props.isDark ? '#666' : '#ccc')};
  background-color: ${(props: CheckboxProps) => (props.isChecked ? '#FF6B6B' : 'transparent')};
  justify-content: center;
  align-items: center;
  margin-right: 8px;
`;

const CheckboxCheck = styled.Text`
  color: white;
  font-size: 14px;
`;

const CheckboxLabel = styled.Text<ThemeProps>`
  flex: 1;
  font-size: 14px;
  color: ${(props: ThemeProps) => (props.isDark ? '#e0e0e0' : '#333')};
`;

interface ButtonProps {
  disabled?: boolean;
}

const SignUpButton = styled.TouchableOpacity<ButtonProps>`
  width: 100%;
  padding: 16px;
  border-radius: 8px;
  background-color: #FF6B6B;
  align-items: center;
  margin-top: 20px;
  opacity: ${(props: ButtonProps) => (props.disabled ? 0.7 : 1)};
`;

const ButtonText = styled.Text`
  color: white;
  font-size: 16px;
  font-weight: bold;
`;

const Divider = styled.View<ThemeProps>`
  flex-direction: row;
  align-items: center;
  width: 100%;
  margin: 30px 0;
`;

const DividerLine = styled.View<ThemeProps>`
  flex: 1;
  height: 1px;
  background-color: ${(props: ThemeProps) => (props.isDark ? '#444' : '#ddd')};
`;

const DividerText = styled.Text<ThemeProps>`
  color: ${(props: ThemeProps) => (props.isDark ? '#888' : '#999')};
  padding: 0 10px;
  font-size: 12px;
`;

const SocialButtonsContainer = styled.View`
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
`;

const SocialButton = styled.TouchableOpacity`
  flex: 0.48;
  padding: 14px;
  border-radius: 8px;
  border: 1px solid #ddd;
  align-items: center;
`;

const SocialButtonText = styled.Text<ThemeProps>`
  font-size: 14px;
  color: ${(props: ThemeProps) => (props.isDark ? '#e0e0e0' : '#333')};
`;

const BottomText = styled.Text`
  margin-top: 30px;
  font-size: 14px;
  color: #999;
`;

const LinkText = styled.Text`
  color: #FF6B6B;
  font-weight: bold;
`;
