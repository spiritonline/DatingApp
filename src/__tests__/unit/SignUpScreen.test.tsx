import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import SignUpScreen from '../../screens/SignUpScreen';
import { Alert } from 'react-native';

// Mock dependencies
const mockNavigate = jest.fn();
const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockNavigate,
      goBack: mockGoBack,
    }),
  };
});

// Mock Firebase auth
jest.mock('../../services/firebase', () => ({
  auth: {},
}));

jest.mock('firebase/auth', () => ({
  createUserWithEmailAndPassword: jest.fn(() => Promise.resolve()),
}));

// Mock Alert
jest.spyOn(Alert, 'alert').mockImplementation(() => {});

describe('SignUpScreen', () => {
  beforeEach(() => {
    mockNavigate.mockClear();
    mockGoBack.mockClear();
    jest.clearAllMocks();
  });

  it('renders correctly with all form fields', () => {
    const { getByText, getByTestId } = render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    
    expect(getByText('Create Account')).toBeTruthy();
    expect(getByTestId('email-input')).toBeTruthy();
    expect(getByTestId('password-input')).toBeTruthy();
    expect(getByTestId('confirm-password-input')).toBeTruthy();
    expect(getByTestId('terms-checkbox')).toBeTruthy();
    expect(getByTestId('location-checkbox')).toBeTruthy();
    expect(getByTestId('signup-button')).toBeTruthy();
  });

  it('shows validation errors when form is submitted with empty fields', () => {
    const { getByText: getByTextLocal, getByTestId } = render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    
    const signUpButton = getByTestId('signup-button');
    fireEvent.press(signUpButton);
    
    expect(getByTextLocal('Email is required')).toBeTruthy();
    expect(getByTextLocal('Password is required')).toBeTruthy();
    expect(getByTextLocal('You must accept the terms and location checks')).toBeTruthy();
  });

  it('shows validation error when passwords do not match', () => {
    const { getByText, getByTestId } = render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    
    const emailInput = getByTestId('email-input');
    const passwordInput = getByTestId('password-input');
    const confirmPasswordInput = getByTestId('confirm-password-input');
    const signUpButton = getByTestId('signup-button');
    
    fireEvent.changeText(emailInput, 'test@example.com');
    fireEvent.changeText(passwordInput, 'password123');
    fireEvent.changeText(confirmPasswordInput, 'password456');
    
    fireEvent.press(signUpButton);
    
    expect(getByText('Passwords do not match')).toBeTruthy();
  });

  it('navigates to MainFeed after successful sign up', async () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    
    const emailInput = getByTestId('email-input');
    const passwordInput = getByTestId('password-input');
    const confirmPasswordInput = getByTestId('confirm-password-input');
    const termsCheckbox = getByTestId('terms-checkbox');
    const locationCheckbox = getByTestId('location-checkbox');
    const signUpButton = getByTestId('signup-button');
    
    fireEvent.changeText(emailInput, 'test@example.com');
    fireEvent.changeText(passwordInput, 'password123');
    fireEvent.changeText(confirmPasswordInput, 'password123');
    fireEvent.press(termsCheckbox);
    fireEvent.press(locationCheckbox);
    
    fireEvent.press(signUpButton);
    
    await waitFor(() => {
      expect(mockNavigate).toHaveBeenCalledWith('MainFeed');
    });
  });

  it('navigates back when back button is pressed', () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    
    const backButton = getByTestId('back-button');
    fireEvent.press(backButton);
    
    expect(mockGoBack).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
