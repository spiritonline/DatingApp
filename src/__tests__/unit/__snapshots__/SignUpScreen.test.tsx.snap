// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpScreen matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  isDark={false}
  style={
    {
      "backgroundColor": "#ffffff",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
          "width": "100%",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 16,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityHint="Go back to previous screen"
            accessibilityLabel="Back"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
            testID="back-button"
          >
            <Text
              isDark={false}
              style={
                {
                  "color": "#333",
                  "fontSize": 24,
                }
              }
            >
              ‚Üê
            </Text>
          </View>
          <Text
            isDark={false}
            style={
              {
                "color": "#333",
                "fontSize": 22,
                "fontWeight": "bold",
              }
            }
          >
            Create Account
          </Text>
          <View
            style={
              {
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              {
                "marginBottom": 16,
                "width": "100%",
              }
            }
          >
            <Text
              isDark={false}
              style={
                {
                  "color": "#333",
                  "fontSize": 14,
                  "marginBottom": 8,
                }
              }
            >
              Email
            </Text>
            <TextInput
              accessibilityHint="Enter your email address"
              accessibilityLabel="Email input"
              autoCapitalize="none"
              autoComplete="email"
              isDark={false}
              keyboardType="email-address"
              onChangeText={[Function]}
              placeholder="Enter your email"
              placeholderTextColor="#999"
              style={
                {
                  "backgroundColor": "#f5f5f5",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderColor": "#ddd",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderWidth": 1,
                  "color": "#000000",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "width": "100%",
                }
              }
              testID="email-input"
              value=""
            />
          </View>
          <View
            style={
              {
                "marginBottom": 16,
                "width": "100%",
              }
            }
          >
            <Text
              isDark={false}
              style={
                {
                  "color": "#333",
                  "fontSize": 14,
                  "marginBottom": 8,
                }
              }
            >
              Password
            </Text>
            <TextInput
              accessibilityHint="Create a password with at least 6 characters"
              accessibilityLabel="Password input"
              isDark={false}
              onChangeText={[Function]}
              placeholder="Create a password"
              placeholderTextColor="#999"
              secureTextEntry={true}
              style={
                {
                  "backgroundColor": "#f5f5f5",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderColor": "#ddd",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderWidth": 1,
                  "color": "#000000",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "width": "100%",
                }
              }
              testID="password-input"
              value=""
            />
          </View>
          <View
            style={
              {
                "marginBottom": 16,
                "width": "100%",
              }
            }
          >
            <Text
              isDark={false}
              style={
                {
                  "color": "#333",
                  "fontSize": 14,
                  "marginBottom": 8,
                }
              }
            >
              Confirm Password
            </Text>
            <TextInput
              accessibilityHint="Confirm your password"
              accessibilityLabel="Confirm password input"
              isDark={false}
              onChangeText={[Function]}
              placeholder="Confirm your password"
              placeholderTextColor="#999"
              secureTextEntry={true}
              style={
                {
                  "backgroundColor": "#f5f5f5",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderColor": "#ddd",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderWidth": 1,
                  "color": "#000000",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "width": "100%",
                }
              }
              testID="confirm-password-input"
              value=""
            />
          </View>
          <View
            style={
              {
                "marginBottom": 12,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": false,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID="terms-checkbox"
            >
              <View
                isChecked={false}
                isDark={false}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomLeftRadius": 4,
                    "borderBottomRightRadius": 4,
                    "borderColor": "#ccc",
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 4,
                    "borderTopRightRadius": 4,
                    "borderWidth": 1,
                    "height": 20,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 20,
                  }
                }
              />
              <Text
                isDark={false}
                style={
                  {
                    "color": "#333",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 14,
                  }
                }
              >
                I accept the Terms of Service and Privacy Policy
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "marginBottom": 12,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": false,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID="location-checkbox"
            >
              <View
                isChecked={false}
                isDark={false}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomLeftRadius": 4,
                    "borderBottomRightRadius": 4,
                    "borderColor": "#ccc",
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 4,
                    "borderTopRightRadius": 4,
                    "borderWidth": 1,
                    "height": 20,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 20,
                  }
                }
              />
              <Text
                isDark={false}
                style={
                  {
                    "color": "#333",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 14,
                  }
                }
              >
                I consent to 3-day location checks
              </Text>
            </View>
          </View>
          <View
            accessibilityHint="Create a new account with the provided information"
            accessibilityLabel="Sign Up Button"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF6B6B",
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "marginTop": 20,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
            testID="signup-button"
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Sign Up
            </Text>
          </View>
          <View
            isDark={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 30,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 30,
                "width": "100%",
              }
            }
          >
            <View
              isDark={false}
              style={
                {
                  "backgroundColor": "#ddd",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 1,
                }
              }
            />
            <Text
              isDark={false}
              style={
                {
                  "color": "#999",
                  "fontSize": 12,
                  "paddingBottom": 0,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 0,
                }
              }
            >
              or sign up with
            </Text>
            <View
              isDark={false}
              style={
                {
                  "backgroundColor": "#ddd",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 1,
                }
              }
            />
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              accessibilityLabel="Sign up with Google"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderColor": "#ddd",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderWidth": 1,
                  "flexBasis": 0,
                  "flexGrow": 0.48,
                  "flexShrink": 1,
                  "opacity": 1,
                  "paddingBottom": 14,
                  "paddingLeft": 14,
                  "paddingRight": 14,
                  "paddingTop": 14,
                }
              }
              testID="google-signup-button"
            >
              <Text
                isDark={false}
                style={
                  {
                    "color": "#333",
                    "fontSize": 14,
                  }
                }
              >
                Google
              </Text>
            </View>
            <View
              accessibilityLabel="Sign up with Apple"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderColor": "#ddd",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "borderWidth": 1,
                  "flexBasis": 0,
                  "flexGrow": 0.48,
                  "flexShrink": 1,
                  "opacity": 1,
                  "paddingBottom": 14,
                  "paddingLeft": 14,
                  "paddingRight": 14,
                  "paddingTop": 14,
                }
              }
              testID="apple-signup-button"
            >
              <Text
                isDark={false}
                style={
                  {
                    "color": "#333",
                    "fontSize": 14,
                  }
                }
              >
                Apple
              </Text>
            </View>
          </View>
          <Text
            style={
              {
                "color": "#999",
                "fontSize": 14,
                "marginTop": 30,
              }
            }
          >
            Already have an account?
             
            <Text
              accessibilityHint="Navigate to log in screen"
              accessibilityLabel="Log In"
              accessibilityRole="link"
              onPress={[Function]}
              style={
                {
                  "color": "#FF6B6B",
                  "fontWeight": "bold",
                }
              }
            >
              Log In
            </Text>
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
